examples/chef/common/clusters/switch/SwitchManager.cpp

```diff
@@ -0,0 +1,167 @@
+        uint8_t newPosition = static_cast<uint8_t>(args[0]);
+        mEventHandler->OnInitialPress(endpointId, newPosition);
+    }
+    break;
+    case Events::LongPress::Id: {
+        uint8_t newPosition = static_cast<uint8_t>(args[0]);
+        mEventHandler->OnLongPress(endpointId, newPosition);
+    }
+    break;
+    case Events::ShortRelease::Id: {
+        uint8_t previousPosition = static_cast<uint8_t>(args[0]);
+        mEventHandler->OnShortRelease(endpointId, previousPosition);
+    }
+    break;
+    case Events::LongRelease::Id: {
+        uint8_t previousPosition = static_cast<uint8_t>(args[0]);
+        mEventHandler->OnLongRelease(endpointId, previousPosition);
+    }
+    break;
+    case Events::MultiPressOngoing::Id: {
+        if (args.size() < 2)
+        {
+            ChipLogError(NotSpecified, "MultiPressOngoing to few arguments");
+            return;
+        }
+        uint8_t newPosition                   = static_cast<uint8_t>(args[0]);
+        uint8_t currentNumberOfPressesCounted = static_cast<uint8_t>(args[1]);
+        mEventHandler->OnMultiPressOngoing(endpointId, newPosition, currentNumberOfPressesCounted);
+    }
+    break;
+    case Events::MultiPressComplete::Id: {
+        if (args.size() < 2)
+        {
+            ChipLogError(NotSpecified, "MultiPressComplete to few arguments");
+            return;
+        }
+        uint8_t previousPosition            = static_cast<uint8_t>(args[0]);
+        uint8_t totalNumberOfPressesCounted = static_cast<uint8_t>(args[1]);
+        mEventHandler->OnMultiPressComplete(endpointId, previousPosition, totalNumberOfPressesCounted);
+    }
+    break;
+    default:
+        break;
+    }
+};
+
+const Clusters::Descriptor::Structs::SemanticTagStruct::Type gLatchingSwitch[] = {
+    { .namespaceID = kNamespaceCommonLevel,
+      .tag         = kTagCommonLow,
+      .label       = chip::Optional<chip::app::DataModel::Nullable<chip::CharSpan>>(
+          { chip::app::DataModel::MakeNullable(chip::CharSpan("Low", 3)) }) },
+    { .namespaceID = kNamespaceCommonLevel,
+      .tag         = kTagCommonMedium,
+      .label       = chip::Optional<chip::app::DataModel::Nullable<chip::CharSpan>>(
+          { chip::app::DataModel::MakeNullable(chip::CharSpan("Medium", 6)) }) },
+    { .namespaceID = kNamespaceCommonLevel,
+      .tag         = kTagCommonHigh,
+      .label       = chip::Optional<chip::app::DataModel::Nullable<chip::CharSpan>>(
+          { chip::app::DataModel::MakeNullable(chip::CharSpan("High", 4)) }) }
+};
+
+void emberAfSwitchClusterInitCallback(EndpointId endpointId)
+{
+    ChipLogProgress(Zcl, "Chef: emberAfSwitchClusterInitCallback");
+
+    ChefRpcActionsWorker::Instance().RegisterRpcActionsDelegate(
+        Clusters::Switch::Id, new SwitchActionsDelegate(Clusters::Switch::Id, new SwitchEventHandler()));
+    SetTagList(/* endpoint= */ 1, Span<const Clusters::Descriptor::Structs::SemanticTagStruct::Type>(gLatchingSwitch));
```

Comment:
The `new` done here leak memory on shutdown. Please create static instances and pass pointers here.

Reply:
Done. PTAL